.github/workflows/k8s_tunnel.yml

```diff
@@ -0,0 +1,132 @@
+name: Tunnel to Juju controller via load balancer on k8s
+
+on:
+  pull_request:
+    paths-ignore:
+      - "README.md"
+      - "project-docs/**"
+  push:
+    branches:
+      - "main"
+    paths-ignore:
+      - "README.md"
+      - "project-docs/**"
+
+# Testing only needs permissions to read the repository contents.
+permissions:
+  contents: read
+
+jobs:
+  # Ensure project builds before running testing matrix
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+      - run: go build -v .
```

Comment:
do we need a `go install` somewhere to ensure we're running the new code?

Reply:
Note that this is not a test, but only a workflow to demonstrate that this can be done. I'm not entirely sure when (on what action) this should be ran, but I generally don't care if it's using the tip or the stable. So this might not even depend on the "build" step, actually.

The terraform set up is down below that uses the `hashicorp/setup-terraform@v2` action and IMO the terraform init/plan should retrieve the stable juju provider from the server.